pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1330,42 +1334,99 @@ func doActivateTail(ctx *domainContext, status *types.DomainStatus,
 // throws back error stating that the device is busy if enabling the vlan
 // filtering is tried immediately after bridge creation.
 // We can either loop retrying in zedrouter or enable it when needed in domainmgr
-func enableVlanFiltering(bridgeName string) {
+func enableVlanFiltering(bridgeName string) error {
 	bridge, err := netlink.LinkByName(bridgeName)
 	if err != nil {
 		log.Errorf("enableVlanFiltering: LinkByName failed for %s: %s", bridgeName, err)
-		return
+		return err
 	}
 	if !*bridge.(*netlink.Bridge).VlanFiltering {
 		// Enable VLAN filtering on bridge
 		if err := netlink.BridgeSetVlanFiltering(bridge, true); err != nil {
-			errStr := fmt.Sprintf("enableVlanFiltering on %s failed: %s",
+			err = fmt.Errorf("enableVlanFiltering on %s failed: %w",
 				bridgeName, err)
-			log.Errorf(errStr)
+			log.Error(err)
+			return err
 		}
 	}
+	return nil
 }
 
-func setupVlans(vifList []types.VifInfo) {
+func setupVlans(vifList []types.VifInfo) error {
+	deadline := time.Now().Add(10 * time.Second)
+	const delay = 500 * time.Millisecond
```

Comment:
might make sense to make this a constant variable.

Reply:
decided to leave it inside as it is specific to this function and not to be used anywhere else